### PR TITLE
Camera control

### PR DESCRIPTION
**Pull request by SabrinaTheCookie**
progress #81 

Can Pan by holding mouse/touch and moving the mouse. 

Pretend that releasing over the background doesn't force you out of the selection zoom.

### CODE REVIEW

**Comment by SabrinaTheCookie** [comment]
@baconeta Can you test this? I'm not vibing with it, but it could be workable with some tweaking.

**Comment by baconeta** [comment]
> @baconeta Can you test this? I'm not vibing with it, but it could be workable with some tweaking.

I'll test this as soon as Zelda goes down

**Comment by baconeta** [comment]
I don't hate this. Sensitivity is a little low. The movement is a bit buggy but it's hard to tell actually. I think we won't know for sure if this is weird or if it's us, until we have a 3D stage. @alleReden - how is that getting on?

File: Assets/Scripts/Utils/StSCamera.cs [hunk 1]
```diff
@@ -1,12 +1,175 @@
...
+     private Vector2 panDelta;
+    [SerializeField] private Vector2 panMin;
+    [SerializeField] private Vector2 panMax;
+    [SerializeField] private float panSensitvity;
+    [Header("The amount of time it takes for panning to reset to the target when changing focus.")]
+    [SerializeField] private float panResetDuration = 1;
+    
+    private void OnEnable()
+    {
+        input = FindObjectOfType<PlayerInput>();
```
Comment (baconeta) [inline Assets/Scripts/Utils/StSCamera.cs:50]: Obviously we only want these actions handled or even bound to during chapter play but we can deal with that later

**Review by baconeta — COMMENTED** [review]
Possibly we could revert this PR and leave it in draft while we work on it and test it?

**Comment by baconeta** [comment]
we have a thought - I will try explain it here and it might help us do some more work on this task/ticket

MAIN GAME STAGE SCREEN
-> Selecting a spot opens the selection UI - dims the main stage except for a spotlight on the selected spot
-> Camera straightens up and zooms into that spot
-> Background can no longer be interacted with until the UI is closed (there will be a button or something like that - or simply clicking in the background could be an option we consider)
-> There will be arrows to the edge of screen to allow a player to change which stage spot is highlighted - camera moves accordingly


This above might prevent weird camera bugs from occurring during selection time

**Comment by SabrinaTheCookie** [comment]
Fixed some bugs.

-> Selecting a spot opens the selection UI - dims the main stage except for a spotlight on the selected spot --- 👍 Lets spin lighting and FX into another ticket. 
-> Camera straightens up and zooms into that spot 👍 
-> Background can no longer be interacted with until the UI is closed (there will be a button or something like that - or simply clicking in the background could be an option we consider) 👍 
-> There will be arrows to the edge of screen to allow a player to change which stage spot is highlighted - camera moves accordingly --- TODO, lets split this into another ticket. Its a fairly small one.

**Comment by baconeta** [comment]
For now we can use https://github.com/Baconeta-Studios/setting-the-stage/issues/76 for the spotlight
I'll add the ticket for changing stage spots with the UI button when in zoomed in mode

Reviewing PR now

File: Assets/Scripts/UI/CoreGameplay/Carousel/Carousel.cs [hunk 2]
```diff
@@ -43,6 +43,10 @@ void Start()
 
     public void OpenCarousel(StagePosition activeStagePosition)
     {
+        if (currentStagePosition && activeStagePosition != currentStagePosition)
+        {
+            CloseCarousel();
```
Comment (baconeta) [inline Assets/Scripts/UI/CoreGameplay/Carousel/Carousel.cs:48]: Do we want to return here ?

**Review by baconeta — APPROVED** [review]
So much better. I like the panning. I can't see us ever needing look-aeround views but maybe a pipedream for a much bigger stage some day. For now - this is just so much better.

Happy to merge though, code is pretty solid apart from comments I made.

Minor merge conflicts but nothing worth crying over

**Comment by baconeta** [comment]
I would argue this closes #81 and progresses #74

File: Assets/Scripts/UI/CoreGameplay/Carousel/Carousel.cs (hunk 2 above)
Comment (SabrinaTheCookie) [reply Assets/Scripts/UI/CoreGameplay/Carousel/Carousel.cs:48]: Nope, added a comment.

**Pull request merged**